pkg/sql/catalog/descs/collection_test.go

```diff
@@ -1217,6 +1220,63 @@ SELECT id
 	}))
 }
 
+func TestDescriptorErrorWrap(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	srv, sqlDB, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer srv.Stopper().Stop(ctx)
+	s := srv.ApplicationLayer()
+
+	tdb := sqlutils.MakeSQLRunner(sqlDB)
+	tdb.Exec(t, `CREATE DATABASE db`)
+	tdb.Exec(t, `USE db`)
+	tdb.Exec(t, `CREATE SCHEMA schema`)
+	tdb.Exec(t, `CREATE TABLE db.schema.table()`)
+
+	monitor := mon.NewMonitor(mon.Options{
+		Name:     "test_monitor",
+		Settings: cluster.MakeTestingClusterSettings(),
+	})
+	monitor.Start(ctx, nil, mon.NewStandaloneBudget(1))
+	ba := monitor.MakeBoundAccount()
+
+	execCfg := s.ExecutorConfig().(sql.ExecutorConfig)
+	for _, tc := range []struct {
+		desc        string
+		err         error
+		isAssertion bool
+	}{
+		{"bare error", errors.New("bare error is treated as an assertion"), true},
+		{"out of memory", ba.Grow(ctx, monitor.Limit()), false},
+		{"pgcode error", sqlerrors.NewTransactionCommittedError(), false},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			require.NoError(t, sql.DescsTxn(ctx, &execCfg, func(
+				ctx context.Context, txn isql.Txn, descriptors *descs.Collection,
+			) error {
+				tn := tree.MakeTableNameWithSchema("db", "schema", "table")
+				_, mut, err := descs.PrefixAndMutableTable(ctx, descriptors.MutableByName(txn.KV()), &tn)
+				if err != nil {
+					return err
+				}
+
+				require.False(t, assert.IsAssertionFailure(tc.err))
```

Comment:
i just noticed that we may want to use `errors.HasAssertionFailure`.

- `HasAssertionFailure` checks the error as well as all the wrapped errors.
- the `errors` package is meant to be the public API.

Reply:
Got it! I wasn’t aware of this. I’ll update it across the backports and address the fix in the master branch with my next PR.